apps/funding-platform/src/entry-server/index.ts

```diff
@@ -0,0 +1,177 @@
+import express, { NextFunction, Request, Response } from "express";
+import { AccessTokenService } from "../access-token";
+import { chainIds, getBalanceForAllChains, getProviders } from "../blockchain";
+import { CreateRequest, RequestService } from "../request";
+import { isExpired } from "../utils";
+
+const app = express();
+
+const tokenIsValid =
+  (
+    accessTokenService: AccessTokenService,
+    requestService: RequestService,
+    maxAmountOfTokens: number,
+    requestFunds?: boolean
+  ) =>
+  async (req: Request, res: Response, next: NextFunction) => {
+    const accessTokenHash = req.headers["x-access-token"] as string;
+    if (!accessTokenHash) return res.status(400).json("Missing Access Token");
+    const dbToken = await accessTokenService.getAccessToken(accessTokenHash);
+    if (!dbToken) return res.status(404).json("Access Token does not exist");
+
+    if (isExpired(dbToken.ExpiredAt)) {
```

Comment:
the validation logic here could be seperated in a function and tested individually